### PR TITLE
feat(worker): export GMS host/port/protocol; bump chart to 0.0.42

### PR DESCRIPTION
**Pull request by rajatoss**
## Summary
Export `DATAHUB_GMS_HOST` / `PORT` / `PROTOCOL` env vars from the `datahub-executor-worker` chart so the executor image's startup readiness gate (`wait_for_gms_ready` in `start.sh`) can find GMS in any deployment, with sensible defaults derived from `gms.url`.

## Problem
The executor image's `wait_for_gms_ready` shell function (the Wolfi-image replacement for `dockerize`) reads `DATAHUB_GMS_HOST` / `PORT` / `PROTOCOL`. The chart only exported `DATAHUB_GMS_URL`. With nothing else set, the gate fell back to a hardcoded `http://datahub-gms:8080/health` and timed out after 240s in any namespace where the actual Service is something like `dh-datahub-gms`.

Observed on `f7700a3ff0-dev07` (pr8856 image, chart 0.0.41): worker pods crashlooping with `Waiting for GMS at http://datahub-gms:8080/health (timeout 240s)...` while the actual Service is `dh-datahub-gms`.

## Behavior
Precedence per field (`host`, `port`, `protocol`):

1. **Any of the three explicitly set in values** → use the explicit values; unset fields fall back to literal internal defaults (`datahub-gms` / `8080` / `http`). The trio is treated as "all-or-nothing" — touching one means the user is describing the internal Service, so other defaults are also internal-friendly. This prevents partial-override silent corruption (e.g., setting `host: dh-datahub-gms` against an external LB URL inheriting `port: 443`).
2. **None of the three set, but `gms.url` is set** → derive all three by parsing the URL (sprig `urlParse`). Self-contained installs (`gms.url: http://datahub-gms:8080`) need no extra config.
3. **Nothing set at all** → literal defaults `datahub-gms` / `8080` / `http` — matches the script's previous hardcoded fallback, preserving backward compatibility with chart 0.0.41.

## Verification

Rendered against 16 edge cases (`helm template` locally), covering:

- Backward compat (no URL, no overrides) — literal defaults preserved
- Self-contained internal URL only — chart parses everything
- External LB URL only — chart parses (probe likely 404s at LB root; documents the case)
- LB URL + partial override (only host set) — port/protocol fall back to internal literals, **not** URL-parsed
- All 3 overrides explicit — exact values pass through
- URL with non-standard port — parsed correctly
- URL with query string / userinfo / path / trailing slash — handled cleanly
- Empty/missing URL — falls to literal defaults
- Each explicit value beats URL parsing — precedence verified

## Companion change
`acryldata/datahub-fork#9666` makes `start.sh` also derive these from `DATAHUB_GMS_URL` as a fallback — defense in depth so future regressions can't recur. That change has merged to `acryl-main`; backport to `releases/v1.1.0` is on branch `cherry-pick/9666-to-releases-v1.1.0`.

### CODE REVIEW

File: charts/datahub-executor-worker/templates/workload.yaml [hunk 1]
```diff
@@ -127,6 +127,12 @@ spec:
           env:
             - name: DATAHUB_GMS_URL
               value: {{ ((.Values.global.datahub).gms).url }}
+            - name: DATAHUB_GMS_PROTOCOL
+              value: {{ ((.Values.global.datahub).gms).protocol | default "http" | quote }}
+            - name: DATAHUB_GMS_HOST
+              value: {{ ((.Values.global.datahub).gms).host | default "datahub-gms" | quote }}
+            - name: DATAHUB_GMS_PORT
+              value: {{ ((.Values.global.datahub).gms).port | default 8080 | quote }}
```
Comment (skrydal) [inline charts/datahub-executor-worker/templates/workload.yaml]: This would require customers to specify the values explicitly. Can we not attempt to "dissect" url to get all the values in case fields are not specified?

Comment (rajatoss) [reply charts/datahub-executor-worker/templates/workload.yaml]: updated, parsed from the url, also tested with scripts to ensure no edge case breaks the parsing logic

**Review by skrydal — APPROVED** [review]
I wish we had a way to test this

**Comment by rajatoss** [comment]
I did some testing via scripts with different urls
